src/main/scala/ch/epfl/bluebrain/nexus/kg/indexing/ViewIndexer.scala

```diff
@@ -0,0 +1,78 @@
+package ch.epfl.bluebrain.nexus.kg.indexing
+
+import akka.actor.{ActorRef, ActorSystem}
+import cats.MonadError
+import cats.data.EitherT
+import cats.effect.Timer
+import cats.syntax.all._
+import ch.epfl.bluebrain.nexus.kg.async.Projects
+import ch.epfl.bluebrain.nexus.kg.config.Vocabulary.nxv
+import ch.epfl.bluebrain.nexus.kg.resolve.Resolution
+import ch.epfl.bluebrain.nexus.kg.resources.Rejection.NotFound
+import ch.epfl.bluebrain.nexus.kg.resources.Resources._
+import ch.epfl.bluebrain.nexus.kg.resources.{Event, ProjectRef, Rejection, Repo}
+import ch.epfl.bluebrain.nexus.service.indexer.persistence.SequentialTagIndexer
+import monix.eval.Task
+import monix.execution.Scheduler
+
+/**
+  * Indexes project view events.
+  *
+  * @param projects    the project operations
+  * @param resolution  the function used to derive a resolution for the corresponding project
+  * @param retryConfig configuration for applying retries in case of failures
+  */
+class ViewIndexer[F[_]: Repo: Timer](
+    projects: Projects[F],
+    resolution: ProjectRef => Resolution[F],
+    retryConfig: RetryConfig
+)(implicit F: MonadError[F, Throwable]) {
+
+  /**
+    * Indexes the view which corresponds to the argument event. If the resource is not found, or it's not compatible to
+    * a view the event is dropped silently.
+    *
+    * @param event the event to index
+    */
+  def apply(event: Event): F[Unit] = {
+    val projectRef                  = event.id.parent
+    implicit val res: Resolution[F] = resolution(projectRef)
+
+    val result: EitherT[F, Rejection, Boolean] = for {
+      resource     <- get(event.id).toRight[Rejection](NotFound(event.id.ref))
+      materialized <- materialize(resource)
+      view         <- EitherT.fromOption(View(materialized), NotFound(event.id.ref))
+      applied      <- EitherT.liftF(projects.applyView(projectRef, view))
+    } yield applied
+
+    Retry.retryWithBackOff(result.value, retryConfig).map(_ => ())
```

Comment:
the retry mechanism is defined in the `SequentialtagIndexer` already. We have two now?

Reply:
I haven't noticed that the `startLocal` embeds a retry mechanism already. I'll remove it for now, but I think longer term the implementation based on `Timer` is nicer as it works on every `F` type with any shape, not just `() => Future[A]`.